src/test/java/emissary/place/RefreshableServiceProviderPlaceTest.java

```diff
@@ -0,0 +1,96 @@
+package emissary.place;
+
+import emissary.core.IBaseDataObject;
+import emissary.core.Namespace;
+import emissary.directory.DirectoryEntry;
+import emissary.test.core.junit5.UnitTest;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.annotation.Nullable;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class RefreshableServiceProviderPlaceTest extends UnitTest {
+
+    private static final byte[] cfgData = ("SERVICE_KEY = \"UNKNOWN.TEST_PLACE.ID.http://localhost:8001/RefreshablePlaceTest$6050\"\n" +
+            "KEY_1 = 200").getBytes();
+    private static final byte[] cfgDataReload = ("SERVICE_KEY = \"*.TEST_PLACE.ID.http://localhost:8001/RefreshablePlaceTest$5060\"\n" +
+            "KEY_1 = 300").getBytes();
+
+    @Nullable
+    private RefreshablePlaceTest place = null;
+
+    @Override
+    @BeforeEach
+    public void setUp() throws Exception {
+        InputStream config = new ByteArrayInputStream(cfgData);
+        place = new RefreshablePlaceTest(config, null, "http://localhost:8001/RefreshablePlaceTest");
+    }
+
+    @Override
+    @AfterEach
+    public void tearDown() throws Exception {
+        super.tearDown();
+        assertNotNull(place);
+        place.shutDown();
+        place = null;
+    }
+
+    @Test
+    void testReconfigure() {
```

Comment:
Consider adding another test (or block within this test) that calls `place.refresh()` without first calling `invalidate()`